src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingSearchHelpers.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Navigation;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api
+{
+    internal sealed class UnitTestingSearchQuery
+    {
+        public readonly string FullyQualifiedTypeName;
+        public readonly string? MethodName;
+        public readonly int MethodArity;
+        public readonly int MethodParameterCount;
+
+        public UnitTestingSearchQuery(string fullyQualifiedTypeName, string? methodName, int methodArity, int methodParameterCount)
+        {
+            FullyQualifiedTypeName = fullyQualifiedTypeName;
+            MethodName = methodName;
+            MethodArity = methodArity;
+            MethodParameterCount = methodParameterCount;
+        }
+    }
+
+    internal readonly record struct UnitTestingNavigationOptions(
+        bool PreferProvisionalTab = false,
+        bool ActivateTab = true)
+    {
+        public UnitTestingNavigationOptions()
+            : this(PreferProvisionalTab: false)
+        {
+        }
+    }
+
+    internal readonly struct UnitTestingDocumentSpan
+    {
+        private readonly DocumentSpan _documentSpan;
+
+        internal UnitTestingDocumentSpan(DocumentSpan documentSpan)
+        {
+            _documentSpan = documentSpan;
+        }
+
+        public Document Document => _documentSpan.Document;
+        public TextSpan SourceSpan => _documentSpan.SourceSpan;
+
+        public async Task NavigateToAsync(UnitTestingNavigationOptions options, CancellationToken cancellationToken)
```

Comment:
This API is not needed. We would likely never call this because the navigation would happen via existing DTE-based implementation in TE's UI (which supports all VS languages).

Reply:
the problem with that then is needing to figure out things like line-mappings and the like, which roslyn takes care of for you :)